R/02-filter.R

```diff
@@ -106,10 +106,12 @@ gimap_filter <- function(.data = NULL,
                                                         filter(boolVals == TRUE) %>% #drop rows where boolVals is false, so this leaves only filters that flagged a pgRNA for removal
                                                         select(id, filterName) %>% #drop the boolVals column because don't need it anymore
                                                         group_by(id) %>% #group by the pgRNA constructs
-                                                        summarize(relevantFilters = toString(filterName)), #and make a column that comma separates the relevant filters
+                                                        summarize(relevantFilters = toString(filterName)) #and make a column that comma separates the relevant filters
   
   ## save a list of which pgRNAs have a zero count in all final timepoint replicates. 
   #NOTE these are NOT necessarily filtered out
+  if(is.null(filter_replicates_target_col)){ filter_replicates_target_col <- c((ncol(gimap_dataset$transformed_data$log2_cpm)-2) : ncol(gimap_dataset$transformed_data$log2_cpm))} #last 3 columns of the data
```

Comment:
I would say we don't know for sure that this will be the last three columns of the data and we should make this so it works the same way as the "replicates" column specification I do for the normalization step. But we can merge it this way for now and fix it in a future PR.

Reply:
This is checking if they specified the replicates column, and if they didn't by default is setting it to the last three columns. Same as we did previously